staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -431,7 +431,14 @@ func TestEmptyWatchEventCache(t *testing.T) {
 				if e, a := tt.expectedEvent.Object, event.Object; !apiequality.Semantic.DeepDerivative(e, a) {
 					t.Errorf("Expected: %#v, got: %#v", e, a)
 				}
-			case <-time.After(3 * time.Second):
+			case <-time.After(300 * time.Millisecond):
```

Comment:
I'm afraid of that change - did you run this with `stress`?

Reply:
yes,

```
❯ go run -mod=readonly golang.org/x/tools/cmd/stress ./cacher.test -test.run TestEmptyWatchEventCache
5s: 10 runs so far, 0 failures
10s: 31 runs so far, 0 failures
15s: 50 runs so far, 0 failures
20s: 71 runs so far, 0 failures
25s: 91 runs so far, 0 failures
30s: 113 runs so far, 0 failures
35s: 133 runs so far, 0 failures
40s: 155 runs so far, 0 failures
45s: 176 runs so far, 0 failures
50s: 198 runs so far, 0 failures
55s: 219 runs so far, 0 failures
1m0s: 242 runs so far, 0 failures
1m5s: 262 runs so far, 0 failures
1m10s: 284 runs so far, 0 failures
1m15s: 307 runs so far, 0 failures
1m20s: 328 runs so far, 0 failures
1m25s: 351 runs so far, 0 failures
1m30s: 373 runs so far, 0 failures
1m35s: 393 runs so far, 0 failures
1m40s: 416 runs so far, 0 failures
1m45s: 440 runs so far, 0 failures
1m50s: 461 runs so far, 0 failures
1m55s: 483 runs so far, 0 failures
2m0s: 505 runs so far, 0 failures
2m5s: 527 runs so far, 0 failures
2m10s: 550 runs so far, 0 failures
2m15s: 574 runs so far, 0 failures
2m20s: 593 runs so far, 0 failures
2m25s: 615 runs so far, 0 failures
2m30s: 639 runs so far, 0 failures
2m35s: 661 runs so far, 0 failures
2m40s: 683 runs so far, 0 failures
2m45s: 706 runs so far, 0 failures
```

what is your concern ?

Reply:
historically, too short timeouts were causing a lot of flakiness.

I wouldn't go below 1s at least.

Reply:
there is only one case that expects results which fails immediately on https://github.com/kubernetes/kubernetes/blob/777070c9a5d458cbeac7a624e00317cf0b0aecf5/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L629

the `err` (^) is wrapped into a watcher here
https://github.com/kubernetes/kubernetes/blob/777070c9a5d458cbeac7a624e00317cf0b0aecf5/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L1319

the other cases don't expect any results.

Reply:
> historically, too short timeouts were causing a lot of flakiness.

> I wouldn't go below 1s at least.


this ^^^

Reply:
i still don't share this concern for this particular test. please see my previous comment. 300 ms to receive an event from already populated channel is enough.

Reply:
@p0lyn0mial - I don't think the comment matters - I just don't feel  comfortable with too low timeouts given our experience here

Reply:
> i still don't share this concern for this particular test. please see my previous comment. 300 ms to receive an event from already populated channel is enough.

In a healthy env, sure, even much less is enough, but not necessarily in a heavily overloaded env.

Reply:
updated the code to wait 1s instead of 300ms